blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerBuilder.scala

```diff
@@ -207,7 +207,10 @@ class BlazeServerBuilder[F[_]] private (
     copy(socketAddress = socketAddress)
 
   def withExecutionContext(executionContext: ExecutionContext): BlazeServerBuilder[F] =
-    copy(executionContext = executionContext)
+    withExecutionContextF(executionContext.pure[F])
+
+  def withExecutionContextF(executionContextF: F[ExecutionContext]): BlazeServerBuilder[F] =
```

Comment:
This seems weird to expose.  `flatMap` that shit.  It makes sense internally, since our default value comes from the constraint, but I don't know why anybody would _call_ it that way.

Reply:
Yeah, I guess since the caller of `.resource` / `stream` is likely to be the same as the one who passes the EC, they can `flatMap` that shit themselves.